ml_genn/ml_genn/compilers/event_prop_compiler.py

```diff
@@ -580,9 +574,21 @@ def pre_compile(self, network: Network,
         # Build list of output populations
         readouts = [p for p in network.populations
                     if p.neuron.readout is not None]
-
-        return CompileState(self.losses, readouts,
+        compile_state = CompileState(self.losses, readouts,
                             genn_model.backend_name)
+        # use explicit dictionary of optimisers if provided
+        if "optimisers" in kwargs.keys():
```

Comment:
you don't need the ``keys()``
```suggestion
        if "optimisers" in kwargs:
```